src/OrleansAWSUtils/Reminders/DynamoDBReminderTable.cs

```diff
@@ -0,0 +1,294 @@
+                    $"Intermediate error reading reminder entry {Utils.DictionaryToString(keys)} from table {TABLE_NAME_DEFAULT_VALUE}.", exc);
+                throw;
+            }
+        }
+
+        public async Task<ReminderTableData> ReadRows(GrainReference grainRef)
+        {
+            var expressionValues = new Dictionary<string, AttributeValue>
+                {
+                    { $":{SERVICE_ID_PROPERTY_NAME}", new AttributeValue(serviceId.ToString()) },
+                    { $":{GRAIN_REFERENCE_PROPERTY_NAME}", new AttributeValue(grainRef.ToKeyString()) }
+                };
+
+            try
+            {
+                var expression = $"{SERVICE_ID_PROPERTY_NAME} = :{SERVICE_ID_PROPERTY_NAME} AND {GRAIN_REFERENCE_PROPERTY_NAME} = :{GRAIN_REFERENCE_PROPERTY_NAME}";
+                var records = await storage.ScanAsync(TABLE_NAME_DEFAULT_VALUE, expressionValues, expression,
+                    item => new ReminderEntry
+                    {
+                        ETag = item[ETAG_PROPERTY_NAME].N,
+                        GrainRef = grainRef,
+                        Period = TimeSpan.Parse(item[PERIOD_PROPERTY_NAME].S),
+                        ReminderName = item[REMINDER_NAME_PROPERTY_NAME].S,
+                        StartAt = DateTime.Parse(item[START_TIME_PROPERTY_NAME].S)
+                    }).ConfigureAwait(false);
+
+                return new ReminderTableData(records);
+            }
+            catch (Exception exc)
+            {
+                logger.Warn(ErrorCode.ReminderServiceBase,
+                    $"Intermediate error reading reminder entry {Utils.DictionaryToString(expressionValues)} from table {TABLE_NAME_DEFAULT_VALUE}.", exc);
+                throw;
+            }
+        }
+
+        public async Task<ReminderTableData> ReadRows(uint beginHash, uint endHash)
+        {
+            var expressionValues = new Dictionary<string, AttributeValue>
+                {
+                    { $":{SERVICE_ID_PROPERTY_NAME}", new AttributeValue(serviceId.ToString()) },
+                    { $":Begin{GRAIN_HASH_PROPERTY_NAME}", new AttributeValue { N = beginHash.ToString() } },
+                    { $":End{GRAIN_HASH_PROPERTY_NAME}", new AttributeValue { N = endHash.ToString() } }
+                };
+
+            try
+            {
+                string expression = string.Empty;
+                if (beginHash < endHash)
+                {
+                    expression = $"{SERVICE_ID_PROPERTY_NAME} = :{SERVICE_ID_PROPERTY_NAME} AND {GRAIN_HASH_PROPERTY_NAME} > :Begin{GRAIN_HASH_PROPERTY_NAME} AND {GRAIN_HASH_PROPERTY_NAME} <= :End{GRAIN_HASH_PROPERTY_NAME}";
+                }
+                else
+                {
+                    expression = $"{SERVICE_ID_PROPERTY_NAME} = :{SERVICE_ID_PROPERTY_NAME} AND ({GRAIN_HASH_PROPERTY_NAME} > :Begin{GRAIN_HASH_PROPERTY_NAME} OR {GRAIN_HASH_PROPERTY_NAME} <= :End{GRAIN_HASH_PROPERTY_NAME})";
```

Comment:
Ok... Adding, the index doesn't do the trick: `KeyConditionExpressions must only contain one condition per key`. So is pointless to make the secondary indexes since they can't be used in the Query operation but instead only on Scan. I don't know if there is a benefit on have the them just for scan since it is only used on Query...